x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorFiltering.java

```diff
@@ -194,23 +197,41 @@ public ConnectorFiltering build() {
         }
     }
 
-    public static boolean isDefaultRulePresentInFilteringRules(List<FilteringRule> rules) {
-        FilteringRule defaultRule = getDefaultFilteringRule(null);
-        return rules.stream().anyMatch(rule -> rule.equalsExceptForTimestampsAndOrder(defaultRule));
-    }
-
-    public static FilteringRule getDefaultFilteringRule(Instant timestamp) {
+    public static FilteringRule getDefaultFilteringRule(Instant timestamp, Integer order) {
         return new FilteringRule.Builder().setCreatedAt(timestamp)
             .setField("_")
             .setId("DEFAULT")
-            .setOrder(0)
+            .setOrder(order)
             .setPolicy(FilteringPolicy.INCLUDE)
             .setRule(FilteringRuleCondition.REGEX)
             .setUpdatedAt(timestamp)
             .setValue(".*")
             .build();
     }
 
+    public static List<FilteringRule> sortFilteringRulesByOrder(List<FilteringRule> rules) {
```

Comment:
there is a lot of going on in this function, could you add a `javadoc` to explain the flow?